src/ledger/grainAllocation.js

```diff
@@ -117,34 +117,29 @@ function computeImmediateReceipts(
   budget: G.Grain,
   credHistory: CredHistory
 ): $ReadOnlyArray<GrainReceipt> {
-  if (budget <= G.ZERO || !credHistory.length) {
+  if (budget === G.ZERO) {
     return [];
   }
 
+  if (!credHistory.length) {
+    throw new Error("cannot allocate Grain: credHistory is empty");
+  }
+
   const lastSlice = credHistory[credHistory.length - 1];
 
   const immediateCredMap = lastSlice.cred;
 
   const totalCred = sum(immediateCredMap.values());
   if (totalCred === 0) {
-    return [];
+    throw new Error("cannot allocate Grain: cred sums to 0");
```

Comment:
Should be <= 0 in case there's negative cred?

Reply:
checking the sum wouldn't guarantee that there's no negative cred (e.g. `[1,-2,5]`). However, the `splitBudget` method will error if there's any negative cred. Maybe we should just remove this check?

I'm almost tempted to try to enforce this at the type level, and have a new math util module with:

```js
export opaque type Finite: number = number;
export opaque type FiniteNonNegative: number = number;
```

And associated parsers and converters so we can use them at runtime.
I'm tired of constantly checking params against `NaN, Infinity, -Infinity`, etc.

Curious what @wchargin thinks.